web/packages/teleport/src/Locks/CreateLock.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useRef, useState } from 'react';
+import SlidePanel from 'design/SlidePanel';
+
+import { ArrowBack, Trash } from 'design/Icon';
+import { Alert, Box, ButtonPrimary, Flex, Input, Text } from 'design';
+import { Cell } from 'design/DataTable';
+
+import useStickyClusterId from 'teleport/useStickyClusterId';
+import history from 'teleport/services/history';
+import cfg from 'teleport/config';
+
+import { useLocks } from './useLocks';
+import { StyledTable } from './shared';
+
+import type { Positions } from 'design/SlidePanel/SlidePanel';
+import type { CreateLockData, SelectedLockTarget } from './types';
+import type { ApiError } from 'teleport/services/api/parseError';
+
+type Props = {
+  panelPosition: Positions;
+  setPanelPosition: (Positions) => void;
+  selectedLockTargets: SelectedLockTarget[];
+  setSelectedLockTargets: (lockTargets: SelectedLockTarget[]) => void;
+};
+
+export function CreateLock({
+  panelPosition,
+  setPanelPosition,
+  selectedLockTargets,
+  setSelectedLockTargets,
+}: Props) {
+  const { clusterId } = useStickyClusterId();
+  const { createLock } = useLocks(clusterId);
+  const [error, setError] = useState('');
+
+  const messageRef = useRef<HTMLInputElement>(null);
+  const ttlRef = useRef<HTMLInputElement>(null);
+
+  function handleCreateLock() {
+    setError('');
+    selectedLockTargets.forEach(async lockTarget => {
+      const lockData: CreateLockData = {
+        targets: { [lockTarget.type]: lockTarget.name },
+      };
+      const message = messageRef?.current?.value;
+      const ttl = ttlRef?.current?.value;
+      if (message) lockData.message = message;
+      if (ttl) lockData.ttl = ttl;
+      await createLock(clusterId, lockData)
+        .then(() => {
+          setTimeout(() => {
+            // It takes longer for the cache to be updated when adding locks so
+            // this waits 1s before redirecting to fetch the list again.
+            history.push(cfg.getLocksRoute(clusterId));
+          }, 1000);
```

Comment:
I don't think we block this way anywhere else in the UI do we?

Reply:
We don't, but I also wasn't able to find another instance where an action that was taken in the UI would result in a delay while we waited for cache propagation. 

I'm not too happy with the necessity of this fix... 

The other option would be to inject the new item (or remove the deleted one) from the UI. This however had an issue where the timestamps would differ when the UI updated which felt like it would be a little odd from the user perspective.

Reply:
Will the UI at least stay in some sort of loading state during this time?

Reply:
These are all good points, I'll add a loading state while this timeout is in effect.

Reply:
Loading states have been added.